### ISSUE TITLE
Double start of token ids in TrOCR add auto ?

### ISSUE THREAD

**Issue by Mohammed20201991** [microsoft/unilm#1067]
**Describe the bug**
The model I am using (TrOCR Model):

The problem arises when using:
* [ ] the official example scripts: done by the nice tutorial  [(fine_tune)](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/TrOCR/Fine_tune_TrOCR_on_IAM_Handwriting_Database_using_Seq2SeqTrainer.ipynb) @NielsRogge 
* [ ] my own modified scripts: (as the script below )
* [ ] 
```
# !pip install -q transformers datasets jiwer
import os
os.environ['CUDA_VISIBLE_DEVICES'] = '3'
import pandas as pd
from datasets import load_metric
import evaluate
from evaluate import load
from transformers import Seq2SeqTrainer, Seq2SeqTrainingArguments , VisionEncoderDecoderModel,default_data_collator ,TrOCRProcessor
from sklearn.model_selection import train_test_split
import torch , argparse
from torch.utils.data import Dataset
from PIL import Image

parser = argparse.ArgumentParser(description="Example script for finetuning TrOCR model",
                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
parser.add_argument("text_path", help="Location of transcriptions (single text file)")
parser.add_argument("images_path", help="Location of image files (folder)")
args = parser.parse_args()
config = vars(args)    
 
cer_metric = load("cer")
wer_metric = load("wer")

train_text = config['text_path']
imgs = config['images_path']

processor = TrOCRProcessor.from_pretrained("microsoft/trocr-large-handwritten")#microsoft/trocr-base-handwritten
def compute_metrics(pred):
    labels_ids = pred.label_ids
    print('labels_ids',len(labels_ids), type(labels_ids),labels_ids)
    pred_ids = pred.predictions
    print('pred_ids',len(pred_ids), type(pred_ids),pred_ids)
    pred_str = processor.batch_decode(pred_ids, skip_special_tokens=True)
    print(pred_str)
    labels_ids[labels_ids == -100] = processor.tokenizer.pad_token_id
    label_str = processor.batch_decode(labels_ids, skip_special_tokens=True)
    print(label_str)

    cer = cer_metric.compute(predictions=pred_str, references=label_str) * 100
    wer = wer_metric.compute(predictions=pred_str, references=label_str) * 100
  
    return {"cer": cer, 'wer': wer}

def load_jsonl():
    return pd.read_json(
                        path_or_buf = train_text,
                        lines=True) 

class LaiaDataset(Dataset):
    def __init__(self, root_dir, df, processor, max_target_length=128):
        self.root_dir = root_dir
        self.df = df
        self.processor = processor
        self.max_target_length = max_target_length

    def __len__(self):
        return len(self.df)

    def __getitem__(self, idx):
        # get file name + text 
        file_name = self.df['file_name'][idx]
        text = self.df['text'][idx]
        # prepare image (i.e. resize + normalize)
        image = Image.open(self.root_dir + file_name).convert("RGB")
        pixel_values = self.processor(image, return_tensors="pt").pixel_values
        # add labels (input_ids) by encoding the text
        labels = self.processor.tokenizer(text, 
                                          padding="max_length",
					                      truncation=True, # this is new trying to solve not equal lenths 
                                          max_length=self.max_target_length).input_ids
        # important: make sure that PAD tokens are ignored by the loss function
        labels = [label if label != self.processor.tokenizer.pad_token_id else -100 for label in labels]
        # encoding  
        return {"pixel_values": pixel_values.squeeze(), "labels": torch.tensor(labels)}

def create_datasets(df: pd.DataFrame):
    train_df, val_df = train_test_split(df, test_size=0.3)

    train_df.reset_index(drop=True, inplace=True)
    val_df.reset_index(drop=True, inplace=True)

    train_dataset = LDataset(
        			root_dir=imgs,
        			df=train_df,
        			processor=processor)
    eval_dataset = LDataset(
        			root_dir=imgs,
        			df= val_df,
        			processor=processor)
    return train_dataset, eval_dataset ,train_df

def main():
    df =   load_jsonl()
    print(df.head(2),df.tail(2)) 
    train_dataset, eval_dataset ,train_df = create_datasets(df) 
    print("Number of training examples:", len(train_dataset))
    print("Number of validation examples:", len(eval_dataset))
    # Let's verify an example from the training dataset
    encoding = train_dataset[0]
    for k,v in encoding.items():
       print(k, v.shape)
    # We can also check the original image and decode the labels:          
    image = Image.open(train_dataset.root_dir + train_df['file_name'][0]).convert("RGB")
    print(image)

    labels = encoding['labels']
    labels[labels == -100] = processor.tokenizer.pad_token_id
    label_str = processor.decode(labels, skip_special_tokens=False) #True
    print(label_str)

    FULL_TRAIN = True
    if FULL_TRAIN:
         #Train a model
        model = VisionEncoderDecoderModel.from_pretrained("microsoft/trocr-large-handwritten")
        # set special tokens used for creating the decoder_input_ids from the labels
        model.config.decoder_start_token_id = processor.tokenizer.cls_token_id
        model.config.pad_token_id = processor.tokenizer.pad_token_id
        # make sure vocab size is set correctly
        model.config.vocab_size = model.config.decoder.vocab_size
        # set beam search parameters
        model.config.eos_token_id = processor.tokenizer.sep_token_id
        model.config.max_length = 64 
        model.config.early_stopping = True
        model.config.no_repeat_ngram_size = 3
        model.config.length_penalty = 2.0
        model.config.num_beams = 4

        model.config.decoder.is_decoder = True # ؟
        model.config.decoder.add_cross_attention = True
        working_dir =  './test/'
        training_args = Seq2SeqTrainingArguments(
        		learning_rate=2e-5,
        		num_train_epochs=2, 
                predict_with_generate=True,
                evaluation_strategy="steps",
                per_device_train_batch_size=2,
                per_device_eval_batch_size=2,
                fp16=True, 
                output_dir=f"{working_dir}",
                logging_steps=500, 
                save_steps=10,
                eval_steps=10,
        		save_total_limit = 1,
                load_best_model_at_end =True, 
        		gradient_checkpointing=True, 
        		)
        # instantiate trainer
        trainer = Seq2SeqTrainer(
                model=model,                
                args=training_args,
                train_dataset = train_dataset,
                eval_dataset = eval_dataset,
                compute_metrics=compute_metrics,
                tokenizer = processor.feature_extractor,
                data_collator = default_data_collator,
                                 )
    trainer.train()
if __name__ == '__main__':
    main()

# python3 train.py path/to/labels  path/to/images/
```

**To Reproduce**
1. After training the model or during the training phase  when evaluating metrics calculate I see the model added double start of token <s><s> or ids [0,0, ........................,2,1,1, 1 ]
2. here is an example during the show generated tokens in compute_metrics 
```
Input predictions: [[    0     0   506  4422  8046     2     1     1     1     1     1     ]]
Input references: [[    0   597  2747 ...     1     1     1]]
```
4. Other examples during testing models [as in this image](https://github.com/Mohammed20201991/DataSets/commit/f5f1964dd8aee1db37c385cd709fae4f202972e6)

**Expected behavior**
In 2 reproduced problems: I am expecting during training 
Input predictions: [[    0    506  4422  8046     2     1     1     1     1     1 ]]
In addition during the testing phase:  generated text without double **<s><s>**
tensor([[    0,     11867,   405, 22379,  1277, ..................,   368,     2]]) 
 <s>ennyit erről, tőlem fényképezz amennyit akarsz, a véleményem akkor</s>
- Platform: Linux  Ubuntu  distribution  [GCC 9.4.0] on Linux 
- PyTorch version (GPU?):  0.8.2+cu110 
- transformers: 4.22.2
- Python version:3.8.10  
@riteshKumarUMass @Dod-o @donglixp

**Comment by Mohammed20201991** [comment]
Thanks to Natabara for his comment The solution is super easy by just skipping <s> start token `labels= labels[1:] ` coming from the tokenizer because the tokenizer adds start token <s> and the TrOCR adds start token <s> automatically as mentioned in TrOCR paper 
``` def __getitem__(self, idx): 
# get file name + text file_name = self.df['file_name'][idx] 
text = self.df['text'][idx] 
# prepare image (i.e. resize + normalize) 
image = Image.open(self.root_dir + file_name).convert("RGB") 
pixel_values = self.processor(image, return_tensors="pt").pixel_values 
# add labels (input_ids) by encoding the text labels = self.processor.tokenizer(text, padding="max_length", truncation=True, max_length=self.max_target_length).input_ids # important: make sure that PAD tokens are ignored by the loss function labels = [label if label != self.processor.tokenizer.pad_token_id else -100 for label in labels] labels= labels[1:] return {"pixel_values": pixel_values.squeeze(), "labels": torch.tensor(labels)} ```
